### PR TITLE
Keyword `patternProperties`

### PR DESCRIPTION
**Pull request by AgniveshChaubey**
This PR addresses #65

### CODE REVIEW

**Comment by jviotti** [comment]
I'm was today (it's a public holiday), but thanks for the PRs! I'll take a proper look tomorrow 🙏🏻

File: content/2020-12/applicator/patternProperties.markdown [hunk 1]
```diff
@@ -32,5 +32,128 @@ related:
...
+}
+{{</schema>}}
+
+{{<instance-pass `An object instance with properties matching the regex and conforming to the corresponding schema is valid`>}}
+{ "name": "John Doe", "age": 21 }
+{{</instance-pass>}}
+
+{{<instance-annotation>}}
+{
+  ...
```
Comment (jviotti) [inline content/2020-12/applicator/patternProperties.markdown:67]: As I mentioned in the other PR, let's get rid of the ellipsis

File: content/2020-12/applicator/patternProperties.markdown [hunk 2]
```diff
@@ -32,5 +32,128 @@ related:
...
+  "annotations": {
+    "properties": [ "foo" ]
+  }
+}
+{{</instance-annotation>}}
+
+{{<schema `Schema with 'patternProperties', 'properties' and 'additionalProperties' keyword`>}}
```
Comment (jviotti) [inline content/2020-12/applicator/patternProperties.markdown:119]: Can you add an example where you have an overlap between `properties` and `patternProperties`? i.e. a property from `properties` also matches a pattern from `patternProperties`?

**Comment by AgniveshChaubey** [comment]
Addressed the suggested changes!

File: content/2020-12/applicator/patternProperties.markdown [hunk 3]
```diff
@@ -32,5 +32,194 @@ related:
...
+{{</instance-pass>}}
+
+{{<instance-annotation>}}
+[
+  // ...
+  {
+    "valid": true,
+    "keywordLocation": "/patternProperties",
+    "instanceLocation": "",
+    "annotations": [ "name", "age" ]
```
Comment (jviotti) [inline content/2020-12/applicator/patternProperties.markdown:72]:
```suggestion
    "annotation": [ "name", "age" ]
```

File: content/2020-12/applicator/patternProperties.markdown [hunk 4]
```diff
@@ -32,5 +32,194 @@ related:
...
+{{</instance-pass>}}
+
+{{<instance-annotation>}}
+[
+  // ...
+  {
+    "valid": true,
+    "keywordLocation": "/patternProperties",
+    "instanceLocation": "",
+    "annotations": []
```
Comment (jviotti) [inline content/2020-12/applicator/patternProperties.markdown:104]:
```suggestion
    "annotation": []
```

File: content/2020-12/applicator/patternProperties.markdown [hunk 5]
```diff
@@ -32,5 +32,194 @@ related:
...
+{{<instance-pass `An instance with properties not matching any regex is valid`>}}
+{ "zbaz": "zbaz" }
+{{</instance-pass>}}
+
+{{<instance-annotation>}}
+[
+  // ...
+  {
+    "valid": true,
+    "keywordLocation": "/patternProperties",
```
Comment (jviotti) [inline content/2020-12/applicator/patternProperties.markdown:102]: This is an interesting one. On first sight, I was doubting whether `patternProperties` would produce an empty array annotation or not annotation at all, but you are right. Maybe worth explicitly mentioning this? i.e. that if `patternProperties` does not match anything, it is still expected to produce an empty array annotation?

File: content/2020-12/applicator/patternProperties.markdown [hunk 6]
```diff
@@ -32,5 +32,194 @@ related:
...
+    "instanceLocation": "",
+    "annotations": [ "foo" ]
```
Comment (jviotti) [inline content/2020-12/applicator/patternProperties.markdown:125]:
```suggestion
    "annotation": [ "foo" ]
```

**Comment by AgniveshChaubey** [comment]
Updated!

**Pull request merged**